packages/client/src/pages/manage-users.vue

```diff
@@ -378,10 +389,10 @@ const { createUser } = useAddUserMutation();
 function addNewUser() {
   Dialog.create({
     component: EditUserDetailsDialog,
-  }).onOk(async (payload: RegisterUserPayload) => {
+  }).onOk(async (payload: Extract<UserDialogPayload, { type: "create" }>) => {
```

Comment:
Nice usage, I'd share it on the tips channel

Reply:
Done. I assume you meant the discriminated union handling.

Reply:
Yes, exactly